fastai/nlp.py

```diff
@@ -188,12 +188,12 @@ def __init__(self, df, text_field, col, newline_eos=True, **kwargs):
         super().__init__(examples, fields, **kwargs)
 
     @classmethod
-    def splits(cls, train_df=None, val_df=None, test_df=None, **kwargs):
-        train_data = None if train_df is None else cls(train_df, **kwargs)
-        val_data = None if val_df is None else cls(val_df, **kwargs)
-        test_data = None if test_df is None else cls(test_df, **kwargs)
-
-        return tuple(d for d in (train_data, val_data, test_data) if d is not None)
+    def splits(cls, train_df=None, val_df=None, test_df=None, keep_nones=False, **kwargs):
+        res = (
+            map_none(train_df, partial(cls, **kwargs)),
+            map_none(  val_df, partial(cls, **kwargs)),
+            map_none( test_df, partial(cls, **kwargs)))
```

Comment:
I think only use `map_none` for `test_df` since I don't think we handle the others being None.

Reply:
ha yep, that makes a lot of sense.